CardData.py

```diff
@@ -0,0 +1,330 @@
+MILL_VALUES['Rare'] = {"standard": 20, "premium": 20, "upgrade": 50}
+MILL_VALUES['Epic'] = {"standard": 50, "premium": 50, "upgrade": 80}
+MILL_VALUES['Legendary'] = {"standard": 200, "premium": 200, "upgrade": 120}
+
+CATEGORIES = {
+    "Aedirn": "Aedirn",
+    "Alchemy": "Alchemy",
+    "Ambush": "Ambush",
+    "An_Craite": "An Craite",
+    "Banish_In_Graveyard": "Doomed",
+    "Bear": "Bear",
+    "Beast": "Beast",
+    "Blitz": "Blitz",
+    "Blue_Stripes": "Blue Stripes",
+    "Breedable": "Breedable",
+    "Brokvar": "Brokvar",
+    "Cintra": "Cintra",
+    "Construct": "Construct",
+    "Cursed_One": "Cursed",
+    "Devourer": "Devourer",
+    "Dimun": "Dimun",
+    "Double_Agent": "Double Agent",
+    "Draconid": "Draconid",
+    "Dragon": "Dragon",
+    "Drummond": "Drummond",
+    "Dwarf": "Dwarf",
+    "Dyrad": "Dryad",
+    "Elf": "Elf",
+    "Harpy": "Harpy",
+    "Heymaey": "Haymaey",
+    "Insectoid": "Insectoid",
+    "Kaedwen": "Kaedwen",
+    "Leader": "Leader",
+    "Mage": "Mage",
+    "Medic": "Medic",
+    "Necrophage": "Necrophage",
+    "Non_Decoyable": "Stubborn",
+    "Non_Medicable": "Permadeath",
+    "Officer": "Officer",
+    "Ogroid": "Ogroid",
+    "Organic": "Organic",
+    "Potion": "Potion",
+    "Redania": "Redania",
+    "Regressing": "Regressing",
+    "Relict": "Relict",
+    "Shapeshifter": "Shapeshifter",
+    "Soldier": "Soldier",
+    "Special": "Special",
+    "Specter": "Specter",
+    "Spell": "Spell",
+    "Spy": "Agent",
+    "Support": "Support",
+    "Svalblod": "Svalblod",
+    "Tactic": "Tactic",
+    "Temeria": "Temeria",
+    "Tordarroch": "Tordarroch",
+    "Tuirseach": "Tuirseach",
+    "Vampire": "Vampire",
+    "Vodyanoi": "Vodyanoi",
+    "War_Machine": "Machine",
+    "Weather": "Weather",
+    "Wild_Hunt": "Wild Hunt",
+    "Witcher": "Witcher"
+}
+
+class CardData:
+    def __init__(self, gwentDataHelper):
+        self._helper = gwentDataHelper
+        self.cardTemplates = self._helper.getCardTemplates()
+        self.tooltipData = self._helper.getTooltipData()
+        self.abilityData = self._helper.getCardAbilities()
+        self.cardNames = {}
+        self.tooltips = {}
+        self.flavorStrings = {}
+
+        for locale in GwentUtils.LOCALES:
+            self.cardNames[locale] = self._helper.getCardNames(locale)
+            self.tooltips[locale] = self._helper.getCardTooltips(locale)
+            self.flavorStrings[locale] = self._helper.getFlavorStrings(locale)
+
+    def createCardJson(self, patch):
+        self.patch = patch
+        # Replace with these values {0} : card id, {1} : variation id, {0} : image size
+        self.imageUrl = "https://firebasestorage.googleapis.com/v0/b/gwent-9e62a.appspot.com/o/images%2F" + patch + "%2F{0}%2F{1}%2F{2}.png?alt=media"
+
+        cardData = self._createBaseCardJson()
+
+        # Requires information about other cards, so needs to be done after we have looked at every card.
+        self._evaluateInfoData(cardData)
```

Comment:
My original statement regarding side-effect functions is not addressed by this change.
Functional purity is independent of 'classes'